sonic_data_client/non_db_client.go

```diff
@@ -428,6 +473,10 @@ func (c *NonDbClient) StreamRun(q *queue.PriorityQueue, stop chan struct{}, w *s
 	for _, sub := range subscribe.GetSubscription() {
 		subMode := sub.GetMode()
 		if subMode != gnmipb.SubscriptionMode_SAMPLE {
+			if subMode == gnmipb.SubscriptionMode_ON_CHANGE {
```

Comment:
you need to change the comment of the StreamRun as well: "It supports SAMPLE mode only."